cloud/gcp/network/main.tf

```diff
@@ -0,0 +1,34 @@
+resource "google_compute_network" "vpc_network" {
+  name                    = "vpc-network"
+  description             = "VPC network for the application. Will have 2 public and 2 private subnets in 2 regions." 
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "private_subnet_1" {
+  name          = "private-subnet-1"
+  ip_cidr_range = "10.192.20.0/24"
+  region        = var.region
+  network       = google_compute_network.vpc_network.id
+}
+
+resource "google_compute_subnetwork" "private_subnet_2" {
+  name          = "private-subnet-2"
+  ip_cidr_range = "10.192.21.0/24"
+  region        = var.region
+  network       = google_compute_network.vpc_network.id
+}
+
+resource "google_compute_subnetwork" "public_subnet_1" {
+  name          = "public-subnet-1"
+  ip_cidr_range = "10.192.10.0/24"
+  region        = var.region
+  network       = google_compute_network.vpc_network.id
+}
+
+resource "google_compute_subnetwork" "public_subnet_2" {
+  name          = "public-subnet-2"
+  ip_cidr_range = "10.192.11.0/24"
+  region        = var.region
+  network       = google_compute_network.vpc_network.id
+}
```

Comment:
suggestion (blocking): please use a descriptive name that suggests the purpose of each subnet, rather than numbers

Reply:
question (blocking): Could you elaborate? What other descriptions are required?

Reply:
I don't see how these subnets are used yet, but I'm curious why there are two public and two private. Are some for the app servers while the others for something else?

Reply:
https://github.com/seattle-uat/civiform/blob/main/infra/network.yaml & https://docs.google.com/document/d/1LAXf3llTk5FD2ihJmMthpqFVLXsJug9z/edit#

Both refer to 4 subnets. 2 for the ECS pods and 2 for NAT. I just translated that to GCP

Reply:
Gotcha, please use a descriptive name so it's clear which ones are for ECS pods and which are for NAT. If we're not sure that's the appropriate network topology for use with cloud run or whatever other container hosting tool let's leave these out for now.

Reply:
Yeah. For cloud run and cloudsql we probably wouldn't need subnets at all